pkg/virt-controller/watch/export/export.go

```diff
@@ -445,23 +506,29 @@ func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMach
 			isPopulated, err = ctrl.isPVCPopulated(pvc)
 			if err != nil {
 				return 0, err
-			} else if isPopulated {
-				pod, err = ctrl.createExporterPod(vmExport, pvcs)
+			}
+			if isPopulated {
+				pod, err = ctrl.createPodAndSecret(vmExport, pvcs)
 				if err != nil {
 					return 0, err
-				} else if pod == nil {
-					return 0, nil
-				}
-
-				if err := ctrl.getOrCreateCertSecret(vmExport, pod); err != nil {
-					return 0, err
 				}
 			}
 		}
 	} else {
-		if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+		inUse, err = ctrl.isPVCInUse(vmExport, pvc)
```

Comment:
I'm not sure we want to do this check.  As-is it will be triggered randomly.  We are not watching all pods and triggering reconcile when an exported PVC is referenced in a pod.  And I don't think we care to do that.

Reply:
Well my main idea behind this one is that we are watching VMIs, and if someone has exported a PVC (instead of a VM) that at least when VMs are started this will get triggered.

Reply:
> Well my main idea behind this one is that we are watching VMIs, and if someone has exported a PVC (instead of a VM) that at least when VMs are started this will get triggered.

I don't think that will work as-is.  The VMI watch queues exports that reference a VM.  It does not queue exports with PVC source.  You would have to go through all the PVCs in the VMI and see if there is an export.

Also if you want to support this case, add a functional test.